src/mito2/src/manifest/storage.rs

```diff
@@ -405,9 +417,42 @@ impl ManifestObjectStore {
         let path = self.checkpoint_file_path(version);
         // Due to backward compatibility, it is possible that the user's checkpoint not compressed,
         // so if we don't find file by compressed type. fall back to checkpoint not compressed find again.
-        let checkpoint_data =
-            match self.object_store.read(&path).await {
-                Ok(checkpoint) => {
+
+        // get the stored_checksum, which is in the checkpoint metadata.
+        let metadata_path = self.last_checkpoint_path();
+        let metadata_bytes = match self.object_store.read(&metadata_path).await {
+            Ok(metadata_bytes) => metadata_bytes,
+            Err(e) if e.kind() == ErrorKind::NotFound => return Ok(None), // No metadata found
+            Err(e) => return Err(e).context(OpenDalSnafu),                // Error reading metadata
+        };
+        let metadata = CheckpointMetadata::decode(&metadata_bytes)?;
+
+        let checkpoint_data = match self.object_store.read(&path).await {
+            Ok(checkpoint) => {
+                // if the checksum exists, examine it.
+                if metadata.checksum != None {
+                    let calculated_checksum = get_checksum(&checkpoint);
+
+                    // verification process : succeed or not.
+                    if Some(format!("{}", calculated_checksum)) == metadata.checksum {
```

Comment:
You could use `ensure` to validate the checksum and return. What's more, I'd prefer `to_string()` here.
```suggestion
                    // verification process : succeed or not.
                    ensure!(checksum == calculated_checksum, VerifyChecksumSnafu { ... });
```